lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -838,72 +848,77 @@ void PatchInOutImportExport::processInOutMessage(CallInst &callInst) {
-    if (callee->isIntrinsic() && callee->getIntrinsicID() == Intrinsic::amdgcn_s_sendmsg) {
-      processInOutMessage(callInst);
+// @param calleeFuncs : a list of candidate callee functions to check
+void PatchInOutImportExport::visitCallInstrs(std::vector<Function *> calleeFuncs) {
+  for (auto callee : calleeFuncs) {
+    auto mangledName = callee->getName();
+
+    if (mangledName.startswith(lgcName::InputImportBuiltIn)) {
+      for (auto user : callee->users()) {
+        if (CallInst *callInst = dyn_cast<CallInst>(user)) {
+          if (callInst->getParent()->getParent() == m_entryPoint)
+            processInputImportBuiltIn(*callInst);
+        }
+      }
+    } else if (mangledName.startswith(lgcName::InputImportGeneric)) {
+      for (auto user : callee->users()) {
+        if (CallInst *callInst = dyn_cast<CallInst>(user)) {
+          if (callInst->getParent()->getParent() == m_entryPoint)
+            processInputImportNonBuiltIn(*callInst, false);
+        }
+      }
+    } else if (mangledName.startswith(lgcName::InputImportInterpolant)) {
+      for (auto user : callee->users()) {
+        if (CallInst *callInst = dyn_cast<CallInst>(user)) {
+          if (callInst->getParent()->getParent() == m_entryPoint)
+            processInputImportNonBuiltIn(*callInst, true);
+        }
+      }
+    } else if (mangledName.startswith(lgcName::OutputImportGeneric)) {
+      for (auto user : callee->users()) {
+        if (CallInst *callInst = dyn_cast<CallInst>(user)) {
+          if (callInst->getParent()->getParent() == m_entryPoint)
+            processOutputImport(*callInst, false);
+        }
+      }
+    } else if (mangledName.startswith(lgcName::OutputImportBuiltIn)) {
+      for (auto user : callee->users()) {
+        if (CallInst *callInst = dyn_cast<CallInst>(user)) {
+          if (callInst->getParent()->getParent() == m_entryPoint)
+            processOutputImport(*callInst, true);
+        }
+      }
+    } else if (mangledName.startswith(lgcName::OutputExportGeneric)) {
+      for (auto user : callee->users()) {
+        if (CallInst *callInst = dyn_cast<CallInst>(user)) {
+          if (callInst->getParent()->getParent() == m_entryPoint)
+            processOutputExportGeneric(*callInst);
+        }
+      }
+    } else if (mangledName.startswith(lgcName::OutputExportBuiltIn)) {
+      for (auto user : callee->users()) {
+        if (CallInst *callInst = dyn_cast<CallInst>(user)) {
+          if (callInst->getParent()->getParent() == m_entryPoint)
+            processOutputExportBuiltIn(*callInst);
+        }
+      }
+    } else if (mangledName.startswith(lgcName::OutputExportXfb)) {
+      for (auto user : callee->users()) {
+        if (CallInst *callInst = dyn_cast<CallInst>(user)) {
+          if (callInst->getParent()->getParent() == m_entryPoint)
+            processOutputExportXfb(*callInst);
+        }
+      }
+    } else if (mangledName.startswith("llvm.amdgcn.s.sendmsg")) {
```

Comment:
I think it is better to check the IntrinsicID instead of doing string comparisons on the name here.